sim-administration/simmanager/src/main/kotlin/org/ostelco/simcards/admin/SimManager.kt

```diff
@@ -42,17 +36,25 @@ object SimManagerSingleton : SimManager {
     }
 
     private fun mapToModelSimEntry(simEntry: org.ostelco.simcards.inventory.SimEntry) : SimEntry {
-        val status = when (simEntry.smdpPlusState) {
-            AVAILABLE, ALLOCATED, CONFIRMED -> SimProfileStatus.NOT_READY
-            RELEASED -> SimProfileStatus.AVAILABLE_FOR_DOWNLOAD
-            DOWNLOADED -> SimProfileStatus.DOWNLOADED
-            INSTALLED -> SimProfileStatus.INSTALLED
-            ENABLED -> SimProfileStatus.ENABLED
-        }
+
+        val status = asSimProfileStatus(simEntry.smdpPlusState)
         return SimEntry(
                 iccId = simEntry.iccid,
                 status = status,
                 eSimActivationCode = simEntry.code ?: "",
                 msisdnList = listOf(simEntry.msisdn))
     }
+
+    fun asSimProfileStatus(smdpPlusState: SmDpPlusState) : SimProfileStatus {
+        return when (smdpPlusState) {
+            SmDpPlusState.AVAILABLE -> SimProfileStatus.NOT_READY
+            SmDpPlusState.ALLOCATED -> SimProfileStatus.NOT_READY
+            SmDpPlusState.CONFIRMED -> SimProfileStatus.NOT_READY
+            SmDpPlusState.RELEASED -> SimProfileStatus.AVAILABLE_FOR_DOWNLOAD
+            SmDpPlusState.DOWNLOADED -> SimProfileStatus.DOWNLOADED
+            SmDpPlusState.INSTALLED -> SimProfileStatus.INSTALLED
+            SmDpPlusState.ENABLED -> SimProfileStatus.ENABLED
+            SmDpPlusState.ALLOCATION_FAILED -> SimProfileStatus.NOT_READY
+        }
```

Comment:
No `else` part? What happens if there are no match, f.ex. in case of incomplete code update?

Reply:
It fails during compilation .-)